src/unit.rs

```diff
@@ -127,8 +130,9 @@ impl<'input, Endian> CompilationUnitHeadersIter<'input, Endian>
         if self.input.is_empty() {
             Ok(None)
         } else {
-            match CompilationUnitHeader::parse(self.input) {
+            match CompilationUnitHeader::parse(self.input, self.offset) {
                 Ok((rest, header)) => {
+                    self.offset.0 += self.input.len() as u64 - rest.len() as u64;
```

Comment:
I wouldn't expect this to affect the bench. I did run the bench, but any change wasn't large enough to be able to measure reliably.